staff/ana-sanjuan/skylab-precourse-projects/final-pasapalabra/app/app.js

```diff
@@ -0,0 +1,239 @@
+        { letter: 'x', answer: 'bótox', status: 0, question: ('CONTIENE LA X. Toxina bacteriana utilizada en cirujía estética') },
+        { letter: 'y', answer: 'peyote', status: 0, question: ('CONTIENE LA Y. Pequeño cáctus conocido por sus alcaloides psicoactivos utilizado de forma ritual y medicinal por indígenas americanos') },
+        { letter: 'z', answer: 'zen', status: 0, question: ('CON LA Z. Escuela de budismo que busca la experiencia de la sabiduría más allá del discurso racional') },
+    ];
+    //initialize counter
+    stop(); 
+    document.getElementById('timeCounter').innerHTML = 150;
+    document.getElementById('pointsCounter').innerHTML = 0;
+    value = 0;
+    timerInterval = setInterval(changeValue, 1000);  
+    
+    //start questions
+    letterIndex = 0, nextWord = []; wordsLeft = 0; counter = 0; hit = 0;
+    console.time('answer time');
+    name = document.getElementById('inputName').value;
+    document.getElementById('instructions').style.visibility = 'hidden'; 
+    document.getElementById('result').innerHTML = 'EMPEZAMOS! 💪';
+    presentQuestion();
+};
+
+function presentQuestion() {
+    if (questions[letterIndex].status === 0){
+        document.getElementById('questions').innerHTML = questions[letterIndex].question;
+        document.getElementById(questions[letterIndex].letter).style.border = ' white solid 4px';
+        controlKey();
+    } else if (counter < 26){
+        letterIndex++;
+        counter++;
+        presentQuestion();
+    } else {
+        startAgain();
+    };
+};
+
+function testResponse() {
+    var word = document.getElementById('response').value;
+    document.getElementById('response').value='';
+    word = word.toLowerCase().trim();
+    if (word === questions[letterIndex].answer){
+        questions[letterIndex].status = 1;
+        document.getElementById('result').innerHTML = 'CORRECTO! 🤩';
+        document.getElementById(questions[letterIndex].letter).style.background = 'radial-gradient(rgb(63, 247, 87),rgb(34, 116, 45)';
+        document.getElementById(questions[letterIndex].letter).style.border = 'white solid 1px';
+        hit++;
+        document.getElementById('pointsCounter').innerHTML = hit;
+    } else if (word === 'pasapalabra'){
+        questions[letterIndex].status = 0;
+        document.getElementById('result').innerHTML = 'PASAPALABRA! 🤔';
+        document.getElementById(questions[letterIndex].letter).style.border = 'white solid 1px';
+    } else {
+        questions[letterIndex].status = 2;
+        document.getElementById('result').innerHTML = 'INCORRECTO! 😣';
+        document.getElementById(questions[letterIndex].letter).style.background = 'radial-gradient(rgb(136, 50, 50), rgb(255, 0, 0))';
+        document.getElementById(questions[letterIndex].letter).style.border = 'white solid 1px';
+    };
+
+    if (counter < 26) {
```

Comment:
this code is not expandible

Reply:
thanks for the comment! what do you mean by expandible?

Reply:
you're comparing a variable with a fixed number, if the number of questions change, the program will stop working.

a better option is to use the array.length variable so you don't have to worry about that

Reply:
thanks for the explanation! makes a lot of sense!